storage/statedb/hasher.go

```diff
@@ -114,6 +114,45 @@ func (h *hasher) hash(n node, db *Database, force bool) (node, node) {
 	return hashed, cached
 }
 
+func (h *hasher) hashRoot(n node, db *Database, force bool) (node, node) {
+	// If we're not storing the node, just hashing, use available cached data
+	if hash, dirty := n.cache(); hash != nil {
+		if db == nil {
+			return hash, n
+		}
+		if !dirty {
+			switch n.(type) {
+			case *fullNode, *shortNode:
+				return hash, hash
+			default:
+				return hash, n
+			}
+		}
+	}
+	// Trie not processed yet or needs storage, walk the children
+	collapsed, cached := h.hashChildren(n, db)
```

Comment:
```suggestion
	collapsed, cached := h.hashChildrenFromRoot(n, db)
```
maybe? I don't find a call site of hashChildrenFromRoot()